Src/FluentAssertions/Xml/XDocumentAssertions.cs

```diff
@@ -344,6 +344,219 @@ public XDocumentAssertions(XDocument document)
         return new AndWhichConstraint<XDocumentAssertions, IEnumerable<XElement>>(this, xElements);
     }
 
+    /// <summary>
+    /// Asserts that the <see cref="XDocument.Root"/> of the current <see cref="XDocument"/> doesn't have the specified child elements.
+    /// </summary>
+    /// <param name="unexpectedElement">
+    /// The name of the expected child element of the current element's <see cref="XDocument"/>.
+    /// </param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public AndConstraint<XDocumentAssertions> NotHaveElement(string unexpectedElement,
+        [StringSyntax("CompositeFormat")] string because = "", params object[] becauseArgs)
+    {
+        Guard.ThrowIfArgumentIsNull(unexpectedElement, nameof(unexpectedElement));
+
+        return NotHaveElement(XNamespace.None + unexpectedElement, because, becauseArgs);
+    }
+
+    /// <summary>
+    /// Asserts that the <see cref="XDocument.Root"/> of the current <see cref="XDocument"/> doesn't have the specified child elements.
+    /// </summary>
+    /// <param name="unexpectedElement">
+    /// The full name <see cref="XName"/> of the expected child element of the current element's <see cref="XDocument"/>.
+    /// </param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public AndConstraint<XDocumentAssertions> NotHaveElement(XName unexpectedElement,
+        [StringSyntax("CompositeFormat")] string because = "", params object[] becauseArgs)
+    {
+        Guard.ThrowIfArgumentIsNull(unexpectedElement, nameof(unexpectedElement));
+
+        bool success = Execute.Assertion
+            .ForCondition(Subject is not null)
+            .BecauseOf(because, becauseArgs)
+            .FailWith(
+                "Did not expect {context:subject} to have an element {0}{reason}, but the element itself is <null>.",
+                unexpectedElement.ToString());
+
+        if (success)
+        {
+            Execute.Assertion
+                .WithExpectation("Did not expect {context:subject} to have an element {0}{reason}, ",
```

Comment:
♻️ You don't need to use `WithExpectation` unless you want to chain multiple `FailWith` calls.